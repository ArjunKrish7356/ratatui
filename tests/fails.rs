use ratatui_macros::constraints;

fn test() {
  // constraints![== 3, == 4; 10];
}
