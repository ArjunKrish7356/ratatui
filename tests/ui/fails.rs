use ratatui_macros::constraints;

fn test() {
  constraints![,];
}

fn test2() {
  let [a, b] = constraints![ == 1/2, == 2, ];
  assert_eq!(a, Constraint::Ratio(1, 2));
  assert_eq!(b, Constraint::Length(2));
}
