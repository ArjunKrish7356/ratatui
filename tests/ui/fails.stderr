error: No rules expected the token `,` while trying to match the end of the macro
 --> tests/ui/fails.rs:5:3
  |
5 |   constraints![,];
  |   ^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::constraints` which comes from the expansion of the macro `constraints` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of macro invocation
  --> tests/ui/fails.rs:8:16
   |
8  |     let [a, b] = constraints![
   |  ________________^
9  | |     == 1/2,
10 | |     == 2,
11 | |   ];
   | |___^ missing tokens in macro arguments
   |
note: while trying to match `==`
  --> src/layout.rs
   |
   |   ( == $token:tt % ) => {
   |     ^^
   = note: this error originates in the macro `$crate::constraints` which comes from the expansion of the macro `constraints` (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `;`
  --> tests/ui/fails.rs:15:51
   |
15 |   let [a, b, c] = constraints![ == 1, == 10%, == 2; 4];
   |                                                   ^ no rules expected this token in macro call
   |
note: while trying to match `%`
  --> src/layout.rs
   |
   |   ( == $token:tt % ) => {
   |                  ^

error[E0527]: pattern requires 2 elements but array has 3
 --> tests/ui/fails.rs:8:7
  |
8 |   let [a, b] = constraints![
  |       ^^^^^^ expected 3 elements
